onnx/backend/test/runner/__init__.py

```diff
@@ -298,17 +299,64 @@ def device_test_func(*args: Any, **kwargs: Any) -> Any:
         for device in devices:
             add_device_test(device)
 
+    @staticmethod
+    def generate_random_data(
+        x: ValueInfoProto, seed: int = 0, name: str = ""
+    ) -> np.ndarray:
+        """
+        Generates a random tensor based on the input definition.
+        """
+        if not x.type.tensor_type:
+            raise NotImplementedError(
+                f"Unable to generate random data for model {name!r} and input {x}."
+            )
+        if x.type.tensor_type.elem_type != 1:
+            raise NotImplementedError(
+                f"Unable to generate random data for model {name!r} and input {x}."
```

Comment:
May be useful to add reason: e.g., "Currently limited to float tensors".

Reply:
done